projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -12,36 +12,33 @@ import {
 } from './cartesian-interval-control.component';
 import { CartesianSummaryComponent, SUMMARIES_DATA } from './cartesian-summary.component';
 
-export class CartesianLegend {
+export class CartesianLegend<TData> {
   private static readonly CSS_CLASS: string = 'legend';
   private static readonly RESET_CSS_CLASS: string = 'reset';
   private static readonly DEFAULT_CSS_CLASS: string = 'default';
   private static readonly ACTIVE_CSS_CLASS: string = 'active';
   private static readonly INACTIVE_CSS_CLASS: string = 'inactive';
   public static readonly CSS_SELECTOR: string = `.${CartesianLegend.CSS_CLASS}`;
 
-  public readonly activeSeries$: Observable<Series<{}>[]>;
-  private readonly activeSeriesSubject: Subject<void> = new Subject();
-  private readonly initialSeries: Series<{}>[];
+  public readonly activeSeries$: Observable<Series<TData>[]>;
+  private readonly activeSeriesSubject: Subject<Series<TData>[]> = new Subject();
+  private readonly initialSeries: Series<TData>[];
 
   private isSelectionModeOn: boolean = false;
   private legendElement?: HTMLDivElement;
-  private activeSeries: Series<{}>[];
+  private activeSeries: Series<TData>[];
   private intervalControl?: ComponentRef<unknown>;
   private summaryControl?: ComponentRef<unknown>;
 
   public constructor(
-    private readonly series: Series<{}>[],
+    private readonly series: Series<TData>[],
     private readonly injector: Injector,
     private readonly intervalData?: CartesianIntervalData,
     private readonly summaries: Summary[] = []
   ) {
     this.activeSeries = [...this.series];
     this.initialSeries = [...this.series];
-    this.activeSeries$ = this.activeSeriesSubject.asObservable().pipe(
-      startWith(undefined),
-      switchMap(() => of(this.activeSeries))
-    );
+    this.activeSeries$ = this.activeSeriesSubject.asObservable().pipe(distinctUntilChanged(), startWith(this.series));
```

Comment:
This distinctUntilChanged doesn't solve our problem - see my previous comment about `isEqual` https://github.com/hypertrace/hypertrace-ui/pull/1270/files/d9be3992407b4436a1c84bc7b826c49942eeb888#r760277561

Reply:
This is not working.
The problem I found it, distinctUntilChanged is not behaving the way we want. I tried to do this, to check where is the problem.

```
this.activeSeries$ = this.activeSeriesSubject.asObservable().pipe(
      tap(console.log),
      distinctUntilChanged((prev, curr) => {
        console.log(prev, curr);
        return prev.length === curr.length;
      })
    );
```
The problem I found was, `prev` is always the first one.

I think, we can take this later as you mentioned, there are only two case where we would need this, only one series (this has already been taken care of where we disable the click behaviour), second when everything is selected and and then we do a reset. I think this is fine, It is also very common to do, so we can skip